### ISSUE TITLE
Support compile-time DI

### ISSUE THREAD

**Issue by julienrf** [flyway/flyway-play#15]
The best practice to ease compile-time DI is to provide a trait named after the name of your module and ending with `Components` and providing a lazy val instantiating the module.

**Comment by gavares** [comment]
+1 for this.

In the meantime, is the proper way to apply db evolutions in my custom AppLoader something like:
```scala
new PlayInitializer(configuration, environment, webCommands)
```

**Comment by jacobono** [comment]
:+1: for this as well.  I'd be happy to submit a PR if you would like :wink: 

thanks @gavares for that snippet. It worked for me in the meantime.

**Comment by gmalouf** [comment]
+1 for this - would play much nicer with modern play apps

**Comment by tototoshi** [comment]
Thank you all. I just released flyway-play 3.0.0. This release supports compile-time DI. Please try it.

https://github.com/flyway/flyway-play/blob/3.0.0/playapp/app/loader/MyApplicationLoader.scala

**Comment by julienrf** [comment]
Looks good :)
